core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateProviderImpl.java

```diff
@@ -260,12 +252,13 @@ public void run() {
                }
                try {
                   Flowable.fromPublisher(stProvider.entryPublisher(k -> shouldSendKey(k) && !dataContainer.containsKey(k), true, true))
-                        .map(XSiteState::fromCacheLoader)
-                        .takeUntil(l -> canceled)
-                        .buffer(chunkSize <= 0 ? Integer.MAX_VALUE : chunkSize)
-                        // We want the CacheException to be thrown to the catch block
-                        .blockingForEach(l -> {
+                          .map(XSiteState::fromCacheLoader)
+                          .takeUntil(l -> canceled)
+                          .buffer(chunkSize)
+                          // We want the CacheException to be thrown to the catch block
+                          .blockingForEach(l -> {
```

Comment:
Just an FYI, but this is actually blocking irrespective of the Consumer, due to the persistence manager not returning entries in the invoking thread. Just FYI, we may want to comment that so it isn't forgotten for future.

Reply:
Yeah, I didn't intend to change anything about xsite in this PR.